privitize attributes

bcoca opened the pull request:
##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
playbooks

bcoca committed:
privitize attributes

bcoca committed:
itemize

bcoca committed:
remove evil extra whitespace that causes apocalypse

bmillemathias commented on changelogs/fragments/private_really.yml:
```diff
@@ -0,0 +1,2 @@
+bugfixes:
+  - ensure private attributes stay private.
```
looking to files modified I assume this is only for playbook (I did not know playbook had this), if this is the case, that could be added in the sentence.

bcoca replied:
yep, only playbook objects have attributes .. but only very few devs actually know that, i'll update

bcoca committed:
btr clog

bcoca committed:
unholy spaces sent to the depths of tarterus

bcoca committed:
also fixed suppresion msg

  was being triggered on all parser errors, not just invalid attribute

bcoca committed:
move private validation, ensure we always do first

  also moved type check to common function

bcoca committed:
simplified block and task_include

 - simpler block logic, but need to revisit why dict/list can be outputed
 - removed preprocess from task_include, moved 'attribute validation' to
   _validate_attributes as is done in other playbook objects.

bcoca committed:
simpler is_block

bcoca committed:
added commentary for future bcoca not to freak

bcoca committed:
killed the rebelious blank line that threatened us

bcoca committed:
no frezing since some are not strings?

bcoca committed:
blocks might not be blocks, skip validate if so

bcoca committed:
missed one

bcoca committed:
added missing private

bcoca committed:
restore optimization, add note

bcoca committed:
added more priv attributes

 better err msg

bcoca committed:
fypoes

bcoca committed:
add context

bcoca committed:
de F it

bcoca committed:
moar private

bcoca committed:
fix name and remove unused import

samdoran commented:
!waffling core_review

bcoca committed:
all subclassers also got private

samdoran commented:
!waffling needs_revision

bcoca committed:
ensure always dict

bcoca committed:
all vliad

bcoca committed:
super ds

bcoca committed:
reverse course

bcoca committed:
hopefully fix none issue and more correct isinst

bcoca committed:
coooomaaaaa!!!!

bcoca committed:
missing clean

bcoca committed:
test

bcoca committed:
fixed blockage

bcoca committed:
nicer restructure
